iscsi.go

```diff
@@ -234,19 +275,26 @@ type Read16 struct {
 	BlockSize int
 }
 
-func (d *device) Read16(data Read16) ([]byte, error) {
+func (d *device) Read16(ctx context.Context, data Read16) ([]byte, error) {
```

Comment:
Not sure I completed follow the addition of `context.Context`.  It seems like it allows interrupting the task polling but that doesn't stop libiscsi from continuing to work on that task the next time anything polls for events on the fd.  Can we expect that callers manage timeouts themselves since libiscsi isn't aware of the timeout and may still complete the operation?

Reply:
Found some TODO comments about `iscsi_set_timeout`, that might be a solution if we want to get libiscsi to give up on a task early but unless timeouts are essential to this PR maybe we could remove the context.Context for now and look into `iscsi_set_timeout` as a separate issue?